clusterloader2/pkg/measurement/common/slos/api_responsiveness_prometheus.go

```diff
@@ -0,0 +1,280 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+TODO(krzysied): This measurement should replace api_responsiveness.go.
+*/
+
+package slos
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	apiResponsivenessPrometheusMeasurementName = "APIResponsivenessPrometheus"
+
+	// latencyQuery %v should be replaced with query window size.
+	latencyQuery = "quantile_over_time(0.99, apiserver:apiserver_request_latency:histogram_quantile[%v])"
+	// countQuery %v should be replaced with query window size.
+	countQuery = "sum(increase(apiserver_request_latency_seconds_count[%v])) by (resource, subresource, scope, verb)"
+
+	latencyWindowSize = 5 * time.Minute
+	queryTimeout      = 5 * time.Minute
+	queryInterval     = 30 * time.Second
+)
+
+func init() {
+	measurement.Register(apiResponsivenessPrometheusMeasurementName, createAPIResponsivenessPrometheusMeasurement)
+}
+
+func createAPIResponsivenessPrometheusMeasurement() measurement.Measurement {
+	return &apiResponsivenessMeasurementPrometheus{}
+}
+
+type apiResponsivenessMeasurementPrometheus struct {
+	startTime time.Time
+	apiCalls  map[string]*apiCall
+}
+
+func (a *apiResponsivenessMeasurementPrometheus) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	var summaries []measurement.Summary
+
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return summaries, err
+	}
+
+	switch action {
+	case "start":
+		a.start()
+	case "gather":
+		summary, err := a.gather(config.ClientSets.GetClient())
```

Comment:
Using the ClientSets from the config won't work for kubemark clusters.
We should be using clientset provided by PrometheusController, introduced in https://github.com/kubernetes/perf-tests/pull/447

Other issue here is to make sure it won't break when Prometheus is not enabled. 

Let's discuss it in person.

Reply:
Assuming that this required for kubemark tests to work, I'm putting this PR on hold.